frontend/packages/dev-console/src/components/topology/__tests__/TopologyPage.spec.tsx

```diff
@@ -0,0 +1,95 @@
+  beforeEach(() => {
+    topologyProps = {
+      activeApplication: 'topology-test',
+      knative: false,
+      cheURL: 'che.openshift.io',
+      serviceBinding: false,
+      match: {
+        params: {
+          name: 'topology-test',
+        },
+        isExact: true,
+        path: '',
+        url: '',
+      },
+    };
+
+    renderTopologyProps = {
+      data: topologyData,
+      loaded: true,
+      loadError: '',
+      serviceBinding: false,
+    };
+  });
+
+  it('should render topology list page', () => {
+    topologyProps.match.path = '/list';
+    const wrapper = shallow(<TopologyPage {...topologyProps} />);
+    expect(wrapper.find(AsyncComponent).exists()).toBe(true);
+  });
+
+  it('should render topology graph page', () => {
+    const wrapper = shallow(<TopologyPage {...topologyProps} />);
+    expect(wrapper.find(ConnectedTopologyDataController).exists()).toBe(true);
+  });
+
+  it('should render projects list page', () => {
+    topologyProps.match.params.name = '';
+    const wrapper = shallow(<TopologyPage {...topologyProps} />);
+    expect(wrapper.find(ProjectListPage).exists()).toBe(true);
+  });
+
+  it('should render view shortcuts button on topology page toolbar', () => {
+    const wrapper = shallow(<TopologyPage {...topologyProps} />);
+    const namespacesPageWrapper = wrapper.find(NamespacedPage).shallow();
+    expect(namespacesPageWrapper.find(Button).props().children).toEqual('View shortcuts');
+  });
+
+  it('should not render view shortcuts button on topology list page toolbar', () => {
+    topologyProps.match.path = '/list';
+    const wrapper = shallow(<TopologyPage {...topologyProps} />);
+    const namespacesPageWrapper = wrapper.find(NamespacedPage).shallow();
+    expect(namespacesPageWrapper.find(Button).exists()).toBe(false);
+  });
+
+  it('should show the topology icon when on topology list page', () => {
+    topologyProps.match.path = '/list';
+    const wrapper = shallow(<TopologyPage {...topologyProps} />);
+    const namespacesPageWrapper = wrapper.find(NamespacedPage).shallow();
+    expect(namespacesPageWrapper.find(Tooltip).props().content).toBe('Topology View');
+    expect(namespacesPageWrapper.find(Link).props().to).toBe('/topology/ns/topology-test');
+  });
+
+  it('should show the topology list icon when on topology page', () => {
+    const wrapper = shallow(<TopologyPage {...topologyProps} />);
+    const namespacesPageWrapper = wrapper.find(NamespacedPage).shallow();
+    expect(namespacesPageWrapper.find(Tooltip).props().content).toBe('List View');
+    expect(namespacesPageWrapper.find(Link).props().to).toBe('/topology/ns/topology-test/list');
+  });
+
+  it('should render topology when workload is loaded', () => {
+    const RenderTopology = React.createElement(renderTopology, renderTopologyProps);
+    const wrapper = shallow(RenderTopology);
+    expect(wrapper.find(StatusBox).exists()).toBe(true);
+    expect(wrapper.find(Topology).exists()).toBe(true);
+  });
+});
```

Comment:
You lack a test for no namespace producing a link to the all-project view page. 

```
                to={`/topology/${namespace ? `ns/${namespace}` : 'all-namespaces'}${
                  showListView ? '' : '/list'
                }`}
```
There are 2 * 2 paths through the link.

Reply:
Added 2 test cases for this.